peerstore.go

```diff
@@ -177,12 +177,13 @@ type peerstore struct {
 	protolock sync.Mutex
 }
 
-// NewPeerstore creates a threadsafe collection of peers.
-func NewPeerstore() Peerstore {
+// NewPeerstore creates a threadsafe collection of peers. ctx restrains the lifetime
+// of automatic expired addresses cleanup. Less frequently used addresses are kept in dstore.
+func NewPeerstore(ctx context.Context, dstore ds.Batching) Peerstore {
```

Comment:
Just to ease integration upwards, can we keep `NewPeerstore` the same, and add a `NewPeerstoreWithDatastore` function that `NewPeerstore` will call with an in-mem map datastore?